pkg/kubelet/kubelet.go

```diff
@@ -2786,6 +2786,17 @@ func (kl *Kubelet) setNodeAddress(node *api.Node) error {
 	return nil
 }
 
+func (kl *Kubelet) updateCloudProviderFromMachineInfo(
+	node *api.Node,
+	info *cadvisorapi.MachineInfo) {
+
+	if info.CloudProvider != cadvisorapi.UnkownProvider &&
+		info.CloudProvider != cadvisorapi.Baremetal {
+		node.Spec.ProviderID = strings.ToLower(string(info.CloudProvider)) +
+			":////" + string(info.InstanceID)
```

Comment:
My lack of cadvisor knowledge shows here.  Is this guaranteed to have a value?  Setting it to `:////` seems less than ideal.